service/src/main/java/bio/terra/catalog/service/dataset/Dataset.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.catalog.service.dataset;
+
+import java.time.Instant;
+import java.util.UUID;
+
+public class Dataset {
+  private UUID id;
+  private String datasetId;
+  private String storageSystem;
```

Comment:
At some layer this will be an enum, I think. If we can make it an enum at this layer, it will probably be better, to avoid having to convert it in a caller method.

Going down to the database, we could make this a constrained field in SQL and only allow valid enum values in the database. I don't feel strongly about that level of validation, but it's possible to do.